File: backend/server/models/post.go
package models

import (
	"gorm.io/gorm"
	"time"
)

type Category struct {
	ID               uint
	ParentCategoryID uint
	Name             string
	ParentCategory   *Category
}

type Post struct {
	gorm.Model
	UserID      uint
	CategoryID  uint
	AddressID   uint
	Title       string
	Description string
	Price       uint
	Status      string
	SoldAt      time.Time
	User        *User
	Category    *Category
	Address     *Address
}

type PostImage struct {
	PostID     uint   `gorm:"primaryKey"`
	ImageURL   string `gorm:"primaryKey"`
	ImageOrder uint
}

File: frontend/ReSell/app/src/main/java/model/Post.kt
package model

import java.time.LocalDateTime
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Post(
    val id: String,
    val userId: String,
    var categoryId: String,
    var title: String,
    var description: String,
    var addressId: String,
    var price: Int,
    var status: String,
    val createdAt: LocalDateTime?,
    var soldAt: LocalDateTime?,
    var updatedAt: LocalDateTime?,
    var deletedAt: LocalDateTime?,
    val conversations: List<Conversation>,
    val order: ShopOrder?,
    val images: List<PostImage>
)
